### PR TITLE
[MM-25659] command to delete channels

### PR DESCRIPTION
**Pull request by ashishbhate**
#### Summary

- Add command to delete channels

#### Ticket Link

- https://mattermost.atlassian.net/browse/MM-25659

#### Related PR
https://github.com/mattermost/mattermost-server/pull/15202

### CODE REVIEW

**Comment by ashishbhate** [comment]
I've updated the server version and the vendor to make it easier to test. I'll revert/update to point to server master before merge.

**Comment by ashishbhate** [comment]
@srkgupta I've added you as QA reviewer here since you're assigned on the server PR.

File: commands/channel.go [hunk 1]
```diff
@@ -573,3 +586,42 @@ func getPrivateChannels(c client.Client, teamID string) ([]*model.Channel, *mode
...
+
+func getChannelDeleteConfirmation() error {
+	var confirm string
+	fmt.Println("Have you performed a database backup? (YES/NO): ")
+	fmt.Scanln(&confirm)
+
+	if confirm != "YES" {
+		return errors.New("aborted: You did not answer YES exactly, in all capitals")
+	}
+	fmt.Println("Are you sure you want to delete the channels specified? All data will be permanently deleted? (YES/NO): ")
```
Comment (isacikgoz) [inline commands/channel.go:598]: Would it make it more smooth to have only one confirmation here, and state both permanent deletion and emphasizing the DB backup?

File: commands/channel_test.go [hunk 2]
```diff
@@ -3044,3 +3044,159 @@ func (s *MmctlUnitTestSuite) TestRemoveChannelUsersCmd() {
...
+	}
+
+	channelName := "channel1"
+	channelID := "channel1Id"
+	mockChannel := model.Channel{
+		Name: channelName,
+		Id:   channelID,
+	}
+
+	s.Run("Delete channels with confirm false returns an error", func() {
```
Comment (isacikgoz) [inline commands/channel_test.go:3063]:
```suggestion
	s.Run("Delete channels without confirm flag returns an error", func() {
```

File: commands/channel.go (hunk 1 above)
Comment (ashishbhate) [reply commands/channel.go:598]: I'm open to changing the confirmation flow. We use similar confirmation text for user and team deletion as well.

Comment (mgdelacroix) [reply commands/channel.go:598]: If we decide to change this, the change should be propagated to keep all the confirmation flows consistent

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1:

File: commands/channel_test.go [hunk 3]
```diff
@@ -3044,3 +3044,159 @@ func (s *MmctlUnitTestSuite) TestRemoveChannelUsersCmd() {
...
+		Name: channelName,
+		Id:   channelID,
+	}
+
+	s.Run("Delete channels without confirm flag returns an error", func() {
+		cmd := &cobra.Command{}
+		cmd.Flags().Bool("confirm", false, "")
+		err := deleteChannelsCmdF(s.client, cmd, []string{"some"})
+		s.Require().NotNil(err)
+		s.Require().Equal(err.Error(), "aborted: You did not answer YES exactly, in all capitals")
```
Comment (mgdelacroix) [inline commands/channel_test.go:3068]: nit: You can merge these two in a single
```go
s.Require().EqualError(err, "aborted: You did not answer YES exactly, in all capitals")
```

**Pull request merged**